src/mlpack/methods/decision_tree/random_binary_numeric_split.hpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file methods/decision_tree/random_binary_numeric_split.hpp
+ * @author Rishabh Garg
+ *
+ * A tree splitter that finds a random binary numeric split.
+ *
+ * @code
+ * @article{10.1007/s10994-006-6226-1,
+ *   author = {Geurts, Pierre and Ernst, Damien and Wehenkel, Louis},
+ *   title = {Extremely Randomized Trees},
+ *   year = {2006},
+ *   issue_date = {April 2006},
+ *   publisher = {Kluwer Academic Publishers},
+ *   address = {USA},
+ *   volume = {63},
+ *   number = {1},
+ *   issn = {0885-6125},
+ *   url = {https://doi.org/10.1007/s10994-006-6226-1},
+ *   doi = {10.1007/s10994-006-6226-1},
+ *   journal = {Mach. Learn.},
+ *   month = apr,
+ *   pages = {3–42},
+ *   numpages = {40},
+ * }
+ * @endcode
```

Comment:
Thanks for adding this.  I think it might be better if you add it in the class comments though---when reading the Doxygen documentation, users are a bit less likely to see file-level documentation. :+1:

Reply:
Sure. 👍